app/models/container_node.rb

```diff
@@ -75,6 +75,12 @@ def ipaddress
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddress, :port => 9090)
+    if MiqRegion.my_region.role_active?("cockpit_ws")
+      uri = URI("/dashboard/cockpit_redirect")
+      uri.query = {:ip => ipaddress }.to_query
+    else
+      uri = URI::HTTP.build(:host => ipaddress, :port => 9090)
```

Comment:
Of course, if the users browser doesn't have ability to access the ip on port 9090 this will not work. My point is that this code is not introduced in the PR. This is how things are currently in master. The point of this PR is to get users to turn on the cockpit worker so that they will not need to be able to access each target server on port 9090.